### ISSUE TITLE
Many users seeing authentication errors on crunchyroll.com

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#31954]
## Description 

Users are having trouble logging in and/or staying logged into crunchyroll.com in Brave. Appears to be cookie related. Some users report that they cannot login at all, others who are able to are then later logged out and when attempting to log back in, see `Something went wrong` error. 

- One user reports that disabling/re-enabling extensions works and that it works in Private windows
- One user states that the site works without logging in but attempting to log in breaks the site
- Site works fine in other Chromium browsers
- Possibly specific to Windows

## Reproduces how often: 

Intermittent

## Brave version (brave://version info)

v.1.56.14

## Version/Channel Information:

- Can you reproduce this issue with the current release? N/A
- Can you reproduce this issue with the beta channel? N/A
- Can you reproduce this issue with the nightly channel? N/A 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No 
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome? No

## Miscellaneous Information:

User reports:
https://community.brave.com/t/crunchy-roll-log-in-not-working-on-brave/495997/19

**Comment by rebron** [comment]
cc: @ryanbr @ShivanKaul

**Comment by ryanbr** [comment]
I don't have a paid account, but tested here: https://gif.fanboy.co.nz/gif/crunchyroll-login.gif

No issues with standard shields here. 

1. Extensions causing issues? (try disabling them all)
2. Clear cookies/cache and relogin
3. Crunchyroll checking for VPN's?
4. Does the login work in private window mode
5. Or a different Brave profile?

**Comment by mobiusevalon** [comment]
After clearing everything except browsing history for all time, disabling shields, and disabling all extensions, I cannot successfully login in normal browser mode. I am not using a VPN at any level. Private mode allows a login with shields and extensions enabled (nearly all of my extensions are also present in private mode,) and other Chromium browsers (Edge, Chrome) can login successfully.

Watched some episodes before work, everything was fine, then come home to see myself logged out and unable to log back in to Crunchyroll.

**Comment by jhunyar** [comment]
I am experiencing this issue as well.

- Latest components
- Shields up or down
- No VPN active or running

![image](https://github.com/brave/brave-browser/assets/1042976/2861ff99-3a7c-4739-a02e-fd4d3a395939)

**Comment by vlokwani** [comment]
I think I'm able to consistently reproduce this issue. For me the **Rakuten** extension is causing it.

Once you're signed out of Crunchyroll.com, if you have the plugin enabled, it won't let you sign back in. The workaround that's consistently working for me is, 
1. disable the Rakuten plugin from brave 
2. delete all crunchyroll.com cookies and stored data in the settings 
3. Reload the crunchyroll.com page.
4. Sign In.

Once you're signed in, you can enable the plugin and use both without any issues till the next time you sign out of Crunchyroll. 

Side Note: I'm only seeing this on my windows machine and not on the MacBook. I have sync turned on across devices.

**Comment by mobiusevalon** [comment]
If I go into Brave Settings under _Privacy and security_ > _Cookies and other site data_ > _See all site data and permissions_, then delete all of Crunchyroll's cookies, I can login again on a non-private window. I can even continue logging out and back in on the primary profile after this operation.

I had previously used _Privacy and security_ > _Clear browsing data_ and deleted everything under _Advanced_ for _All time_ to no effect. I can't be sure if it's intentional or an oversight that managing the site storage more directly deleted whatever bork storage was causing the problem, while what I assumed was the nuclear option did not.

Before deleting Crunchyroll's cookies under _Site data and permissions_, a comparison of the console on a private window (where login was successful) and the primary profile (where it was not) was largely ReCaptcha being blocked and the absence of gstatic.com as a content source on the latter.

**Comment by dillonhafer** [comment]
This is 100% happening on macos. Recaptcha doesn't load and causes the authentication failures. But it worked in private windows. So, I went into "Site Settings > Delete Data" then it started working normally again.

**Comment by sshaffer317** [comment]
Still happening, commenting in hopes of improving visibility. Reinstalled Brave, shields down, cleared cache, deleted site data and permissions, explicitly added crunchyroll.com & sso.crunchyroll.com/login to always allow cookies list. Something clearly changed this summer that just made Crunchyroll shit its pants regularly on Brave. Opera GX it is I guess.

**Comment by ShivanKaul** [comment]
Crunch Roll doesn't use Google Sign In so this is not a Google Sign In Permission related problem. cc @ryanbr I can't create a new account either, it just spins for a while and gives that "Something went wrong" error. This happens even with Shields down.

**Comment by ShivanKaul** [comment]
We're working on a potential fix, stay tuned.

**Comment by ShivanKaul** [comment]
Can folks try the following to help verify our fix:
1. Add `||c.evidon.com/dg/dg.js$script`
2. Go to crunchyroll.com
3. Delete all local storage for sso.crunchyroll.com
4. Try logging in.
![image](https://github.com/brave/brave-browser/assets/5284154/21ddbe15-9211-4bdc-9085-f0be82ef94bf)

**Comment by yjuw** [comment]
@ShivanKaul I had to refresh between steps 3 and 4 to make it work but this did work for me. It's been months since I've been able to use crunchyroll. Thank you!

**Comment by ShivanKaul** [comment]
Thanks, good point, I'll update the steps.

**Comment by MoonshadeSnow** [comment]
I tried the fix @ShivanKaul mentioned and I was able to login like normal but I keep running into a different error Shak-3016. When it happens the show I'm watching crashes and I need to reload the page and it often hops back a few seconds. Sometimes this happens like 6+ times in a 20 minute show so it's not fun to deal with. I got a screeenshot of my Inspect Element Errors. 

![image](https://github.com/brave/brave-browser/assets/148934679/3d8aa1f6-34e5-4c64-a9b8-3ecfa7ae053d)

**Comment by eligrey** [comment]
I can consistently reproduce this issue in any browser with GPC enabled. It's due to their CMP.

**Comment by ShivanKaul** [comment]
@eligrey Yep! Just checking, does the fix in https://github.com/brave/brave-browser/issues/31954#issuecomment-1769442607 work? @ryanbr did we end up rolling out the filter list fix + local storage removal?

**Comment by ryanbr** [comment]
Sorry missed the commit, have merged it now

**Comment by eligrey** [comment]
@ShivanKaul that fix is inapplicable for me. I don't use Brave. Has anyone been able to reach out to Crunchyroll directly to let them know that their site is broken with GPC enabled?

**Comment by ShivanKaul** [comment]
Yes, I submitted a [CPPA](https://cppa.ca.gov/webapplications/complaint) complaint.

**Comment by DannyPoo** [comment]
Hello I am having this issue currently. It works in a private tab mode. I have tried following the steps above, disabled shields, no vpn, or extensions enabled. Here is my console log.
[sso.crunchyroll.com-1702212270577.log](https://github.com/brave/brave-browser/files/13627767/sso.crunchyroll.com-1702212270577.log)

**Comment by ShivanKaul** [comment]
@DannyPoo can you switch on Shields and follow Steps 2 and onwards from https://github.com/brave/brave-browser/issues/31954#issuecomment-1769442607 and let us know if that helps?

**Comment by DannyPoo** [comment]
Switching shields on fixed this issue after following the steps.

> @DannyPoo can you switch on Shields and follow Steps 2 and onwards from [#31954 (comment)](https://github.com/brave/brave-browser/issues/31954#issuecomment-1769442607) and let us know if that helps?

**Comment by ShivanKaul** [comment]
This was a GPC related issue.
